src/main/java/io/confluent/connect/jdbc/dialect/SqlServerDatabaseDialect.java

```diff
@@ -91,6 +94,21 @@ public SqlServerDatabaseDialect(AbstractConfig config) {
     jtdsDriver = jdbcUrlInfo == null ? false : jdbcUrlInfo.subprotocol().matches("jtds");
   }
 
+  public boolean sqlServer2016OrLater() {
```

Comment:
Let's call this by "versionWithBreakingDatetimeChange" or something with more context than just the year of the release. There may be multiple releases in that year, and more directly what the function does.